share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Function aveI(averageOfInteger) takes 2 arguments x and y and adds two values and divide the result by 2. This function shows the other was of defining the parameters using lambda operator. The traditional way of defining the parameter is `def aveI x y = (x+y)/2`. Either of them are logically same.
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR(averageOfReal) takes 2 arguments of type double and divides the result by 2
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveM(Modulus) adds 2 values and provides the remainder 
+```
+def aveM = \x\y (x+y) % 2
+```
+
+* Function duplicate concantenates or copies a given string twice
+```
+def duplicate s  ="{s}{s}"
+```
+
+* integerToUnicode is a `wake built-in function` which takes the ASCII input and provides its equivalent character
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger is a `wake built-in function` which gives the ASCII value of the given character.
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+* strlen is a `wake built-in function` which returns the length of the given string.
+```
+wake -x 'strlen "abcde"'
+```
+
+* `|`is the pipeFn which takes the argument from the left hand side of the `|` and provides it to the right hand side function. This makes the function to be more readable. `(a(b(c(d))))` is represented as `d | c | b | a`. In both the cases d is evaluated first then followed by c , b and a.
+
+### Wake Built-In Functions
+Few of the commonly used built-in functions in this tutorial is given below
+* `explode` breaks the string into a list of string character.
+* `take` takes first `n` elements in the list.
+* `drop` drops first `n` elements in the list.
+* `reverse` reverses the elements in the list.
+* `head` returns the first element in the list.
+* `tail` returns the list by removing the `head`.
+* `cat` concatenates the elements in list of string to a string. 
+* `prepend` places the given data at the begining(head) of the given list
+* `append` appends the given data at the end of the given list  
+
+* Below function takes a string and returns the middle character of the string.
+```
+def middle fullstring =
+  def len = if (strlen fullstring % 2 == 0) then (strlen fullstring / 2) else  ((strlen fullstring / 2) + 1)
+  def outputstr = explode fullstring | take len | reverse | head 
+  outputstr
+```
+
+* Below function takes a string and returns the string removing the first and last character.
+```
+def dtrunc fullstring =
+  def len = strlen fullstring
+  def outputstr = explode fullstring | take (len-1) | tail | cat
+  outputstr
+```
+
+* Function incfirst takes a string and a character string and replaces the first character of the string with the given character string
```

Comment:
what is a "character string"? do you mean "list of characters"?

Reply:
Typo error. It has to be character